### ISSUE TITLE
Global parameters $ref issue

### ISSUE THREAD

**Issue by borisirota** [wcandillon/swagger-js-codegen#41]
Hi,

I'm familiar with the "#/parameters/PARAM_NAME" syntax when accessing globals.
It seems that [the code](https://github.com/wcandillon/swagger-js-codegen/blob/master/lib/codegen.js#L61) expecting it to be without the prefix "#/parameters".

I guess that this prefix is needed because there are few objects that defines global parameters (there is the [definitions object](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#definitionsObject) as well).

I think that the best solution to handle $ref will be something like:
```javascript
var value;
if (obj.$ref.indexOf('#/') === -1) {
    //backwards compatibility
} else {
    var paths  = obj.$ref.split('/');
    var collection = paths[1];
    var key = paths[2];
    value = swagger[collection][key];
}
```

Edit: just noticed that there is no need to access the definition object so I guess the code for the general solution is useless and `value = swagger.parameters[obj.$ref.split('/')[2]];` will be good enough :)

Thanks,
Boris

**Comment by wcandillon** [comment]
@borisirota Can you send me a swagger spec and some operations that should be invoke from this spec? I would like to add it to the test suite and take it from there.

**Comment by borisirota** [comment]
I still didn't start my implementation (just saw this issue in the code) so I generated a valid swagger spec that addresses this issue. The request must contain a query param 'id' which it's schema need to be taken from the global parameters object:
```
swagger: '2.0'

info:
  version: "0.0.0"
  title: <enter your title>

parameters:
  id:
    name: id
    in: query
    description: id
    required: true
    type: string

paths:
  /persons:
    parameters:
      - $ref: "#/parameters/id"
    get:
      description: Gets `Person` object.
      responses:
        200:
          description: Successful response
          schema:
            type: object
            properties:
              name:
                type: string
```

**Comment by borisirota** [comment]
In JSON:
```
{
    "swagger": "2.0",
    "info": {
        "version": "0.0.0",
        "title": "<enter your title>"
    },
    "parameters": {
        "id": {
            "name": "id",
            "in": "query",
            "description": "id",
            "required": true,
            "type": "string"
        }
    },
    "paths": {
        "/persons": {
            "parameters": [
                {
                    "$ref": "#/parameters/id"
                }
            ],
            "get": {
                "description": "Gets `Person` object.",
                "responses": {
                    "200": {
                        "description": "Successful response",
                        "schema": {
                            "type": "object",
                            "properties": {
                                "name": {
                                    "type": "string"
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
```

**Comment by borisirota** [comment]
@wcandillon already commented [here](https://github.com/wcandillon/swagger-js-codegen/pull/45).
In addition, if I understand the code correctly, in the updated code I think that [L45-L50](https://github.com/wcandillon/swagger-js-codegen/blob/master/lib/codegen.js#L45) and [L68](https://github.com/wcandillon/swagger-js-codegen/blob/master/lib/codegen.js#L68) are redundant and wrong:
1. globalParams is an object, not array.
2. no need to concat the globalParams to every operation params because the globalParams can contain params that the current operation doesn't use and only some other operations do use.
3. [L72](https://github.com/wcandillon/swagger-js-codegen/blob/master/lib/codegen.js#L72) already take care of getting the global parameter.

Thanks for the fix !
Boris :)

**Comment by wcandillon** [comment]
@borisirota `globalParams ` might be a confusing name. This variable contains the parameters which are global to the path only (and therefore always included in the method). I don't understand 1. and 2..
1. In the examples you sent me it is an array.
2. `globalParams ` cannot contain parameters that the operation doesn't use.

**Comment by borisirota** [comment]
@wcandillon You are right. I got confused by myself. Sorry :)
So [#45](https://github.com/wcandillon/swagger-js-codegen/pull/45) fixes [#40](https://github.com/wcandillon/swagger-js-codegen/issues/40) and [#41](https://github.com/wcandillon/swagger-js-codegen/issues/41), right?

**Comment by wcandillon** [comment]
correct :)
